File: simple-galaxy/src/main/java/com/galaxy/config/EsConfig.java
package com.galaxy.config;

import org.elasticsearch.client.Client;
import org.elasticsearch.client.transport.TransportClient;
import org.elasticsearch.common.settings.Settings;
import org.elasticsearch.common.transport.InetSocketTransportAddress;
import org.elasticsearch.transport.client.PreBuiltTransportClient;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.elasticsearch.core.ElasticsearchOperations;
import org.springframework.data.elasticsearch.core.ElasticsearchTemplate;
import org.springframework.data.elasticsearch.repository.config.EnableElasticsearchRepositories;

import java.net.InetAddress;

/**
 * Created by wangpeng on 13/03/2018.
 */
@Configuration
@EnableElasticsearchRepositories(basePackages = "com.galaxy.dao")
public class EsConfig {

    @Value("${elasticsearch.host}")
    private String EsHost;

    @Value("${elasticsearch.port}")
    private int EsPort;

    @Value("${elasticsearch.clustername}")
    private String EsClusterName;

    /**
     * @return
     * @throws Exception
     * @see https://www.elastic.co/guide/en/elasticsearch/client/java-api/5.6/transport-client.html
     */
    @Bean
    public Client client() throws Exception {

        Settings settings = Settings.builder()
                .put("cluster.name", EsClusterName)
                .build();

        TransportClient client = new PreBuiltTransportClient(settings).addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(EsHost), EsPort));
        return client;
    }

    @Bean
    public ElasticsearchOperations elasticsearchTemplate() throws Exception {
        return new ElasticsearchTemplate(client());
    }

}

File: elastic-galaxy/src/main/scala/com/galaxy/elastic/conf/EsConfig.scala
package com.galaxy.elastic.conf

import java.net.InetAddress
import java.util.concurrent.TimeUnit

import org.elasticsearch.client.transport.TransportClient
import org.elasticsearch.common.settings.Settings
import org.elasticsearch.common.transport.InetSocketTransportAddress
import org.elasticsearch.common.unit.TimeValue
import org.elasticsearch.transport.client.PreBuiltTransportClient
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.{Bean, ComponentScan, Configuration}
import org.springframework.data.elasticsearch.core.ElasticsearchTemplate
import org.springframework.data.elasticsearch.repository.config.EnableElasticsearchRepositories

@Configuration
@ConfigurationProperties(prefix = "es")
@EnableElasticsearchRepositories(basePackages = Array("com.galaxy.elastic.dao"))
@ComponentScan(basePackages = Array("com.galaxy.elastic.service"))
class EsConfig {

  @Value("${es.host}")
  private var host: String = _

  @Value("${es.port}")
  private var port: Int = 0

  @Bean
  def transportClient: TransportClient = {
    val settings = Settings.builder
      .put("client.transport.ignore_cluster_name", true)
      .put("transport.tcp.connect_timeout", new TimeValue(60, TimeUnit.SECONDS))
      .put("transport.tcp.compress", true)
      .build

    val client = new PreBuiltTransportClient(settings)
    client.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(host), port))
    client
  }

  @Bean
  def elasticsearchTemplate = new ElasticsearchTemplate(transportClient)

}
